### PR TITLE
QA fixes

### PR DESCRIPTION
**Pull request by DhaaraniCIT**
## Summary by CodeRabbit

- **New Features**
	- Introduced new SVG icons for improved visual elements within the app.
	- Enhanced search bar with dynamic width and padding for better user experience.
	- Added a custom checkbox component for a more intuitive configuration interface.
	- Implemented conditional rendering of icons in text fields for better input control.

- **Style**
	- Adjusted padding and margins across various components for a more polished layout.
	- Updated text styles and button designs for consistency and readability.
	- Expanded theming capabilities with additional pixel values for precise design control.

- **Bug Fixes**
	- Corrected the conditional display logic for sub-labels in select fields to ensure accurate information is presented based on context.

- **Documentation**
	- No visible changes to end-users.

- **Refactor**
	- No visible changes to end-users.

- **Tests**
	- No visible changes to end-users.

- **Chores**
	- No visible changes to end-users.

- **Revert**
	- No visible changes to end-users.

### CODE REVIEW

**Comment by DhaaraniCIT** [comment]
@coderabbitai review

File: src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-advanced-settings/sage300-onboarding-advanced-settings.component.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div>
+<div class="tw-pb-48-px">
```
Comment (ashwin1111) [inline src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-advanced-settings/sage300-onboarding-advanced-settings.component.html:1]: fix in qbo, dynamics as well plis

File: src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<div>
+<div class="tw-pb-48-px">
```
Comment (ashwin1111) [inline src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.html:1]: same as above

File: src/app/integrations/sage300/sage300-shared/sage300-advanced-settings/sage300-advanced-settings.component.scss [hunk 3]
```diff
@@ -1,5 +1,5 @@
 .preview-text {
-    @apply tw-text-14-px tw-font-400 tw-text-faded-text-color tw-p-24-px #{!important} ;
+    @apply tw-text-14-px tw-font-400 tw-text-faded-text-color tw-py-24-px tw-pl-62-px tw-pr-24-px #{!important} ;
```
Comment (ashwin1111) [inline src/app/integrations/sage300/sage300-shared/sage300-advanced-settings/sage300-advanced-settings.component.scss:2]: padding top should be 0px, 24px is applied in the upper div as well

File: src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.html [hunk 4]
```diff
@@ -4,7 +4,9 @@
         <div class="fields-header-text tw-w-30-vw">{{brandingConfig.brandName}} Fields</div>
         <div class="fields-header-text tw-flex tw-w-[10vw] tw-whitespace-nowrap">
             Import to {{brandingConfig.brandName}}
-            <svg-icon-sprite src="info" width="20px" hight="20px" class="tw-text-faded-text-color" [pTooltip]="'Importing a dimension syncs active values from ' + appName + ' every 24 hours. Deactivated values in ' + appName + ' will also be disabled in ' + brandingConfig.brandName + ''"></svg-icon-sprite>
+            <svg-icon-sprite src="info" width="20px" hight="20px" class="tw-text-faded-text-color" tooltipPosition="top"
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.html:7]: fix all typo here, height*
Screenshot 2024-01-05 at 8 42 22 PM

File: src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.html [hunk 5]
```diff
@@ -93,33 +96,41 @@ <h5 *ngIf="!isCloneSettingView" class="!tw-text-faded-text-color tw-text-14-px t
...
-                                  {{ option.display_name }}
-                                </div>
-                              </ng-template>
-                        </p-dropdown>
-                        <p-inputSwitch class="tw-pl-32-px input-toggle-section tw-pt-10-px" formControlName="dependentFieldImportToggle" [disabled]="true"></p-inputSwitch>
+                        <div>
+                            <p class="tw-pl-20-px tw-mb-8-px tw-text-14-px tw-text-sub-text-color !tw-font-400">Dependent Field <app-mandatory-field></app-mandatory-field> </p>
+                            <input type="text" [ngClass]="getDynamicWidth(i)" class="tw-text-14-px tw-text-slightly-normal-text-color !tw-ml-20-px !tw-font-500 !tw-h-36-px !tw-px-14-px !tw-py-10-px tw-border tw-border-solid !tw-border-separator tw-rounded-4-px" value="{{dependentField.source_field}}" disabled>
+                        </div>
+                        <svg-icon-sprite src="arrow-line" height="10px" width="100px" class="!tw-ml-0 tw-pt-34-px tw-text-box-color"></svg-icon-sprite>
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.html:113]: allow color doesn't match with design

File: src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.html [hunk 6]
```diff
@@ -93,33 +96,41 @@ <h5 *ngIf="!isCloneSettingView" class="!tw-text-faded-text-color tw-text-14-px t
...
-                                <div [ngClass]="{'custom-option': option.attribute_type === 'custom_field'}">
-                                  <svg-icon-sprite *ngIf="option.attribute_type === 'custom_field'" src="add" width="20px" height="20px" class="icon-align"></svg-icon-sprite>
-                                  {{ option.display_name }}
-                                </div>
-                              </ng-template>
-                        </p-dropdown>
-                        <p-inputSwitch class="tw-pl-32-px input-toggle-section tw-pt-10-px" formControlName="dependentFieldImportToggle" [disabled]="true"></p-inputSwitch>
+                        <div>
+                            <p class="tw-pl-20-px tw-mb-8-px tw-text-14-px tw-text-sub-text-color !tw-font-400">Dependent Field <app-mandatory-field></app-mandatory-field> </p>
+                            <input type="text" [ngClass]="getDynamicWidth(i)" class="tw-text-14-px tw-text-slightly-normal-text-color !tw-ml-20-px !tw-font-500 !tw-h-36-px !tw-px-14-px !tw-py-10-px tw-border tw-border-solid !tw-border-separator tw-rounded-4-px" value="{{dependentField.source_field}}" disabled>
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.html:111]: index would be 0 and 1 always, let's do a simple check directly in the template. i === 0 ? 'tw-w-something' : 'tw-w-somethingelse'

File: src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.html [hunk 7]
```diff
@@ -93,33 +96,41 @@ <h5 *ngIf="!isCloneSettingView" class="!tw-text-faded-text-color tw-text-14-px t
...
-                    </div>
-                    <label class="tw-pl-2 tw-text-14-px" for="checkbox">Import Cost Code and Cost Type from {{appName}} as dependent fields</label>
+                    <label class="container tw-pl-24-px tw-text-14-px">Import Cost Code and Cost Type from {{appName}} as dependent fields
+                        <input type="checkbox" formControlName="isDependentImportEnabled">
+                        <span class="checkmark"></span>
+                    </label>
                 </div>
                 <div *ngIf="form.value.isDependentImportEnabled && dependentImportFields">
-                    <div *ngFor="let dependentField of dependentImportFields" class="tw-flex tw-pt-4">
+                    <div *ngFor="let dependentField of dependentImportFields; index as i" class="tw-flex tw-items-center tw-pt-4" [ngClass]="getDynamicPadding(i)">
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.html:105]: same as below comment

File: src/app/shared/components/configuration/configuration-text-field/configuration-text-field.component.html [hunk 8]
```diff
@@ -7,7 +7,11 @@
     </div>
 
     <div class="tw-mt-8-px">
-        <input [type]="type" class="tw-border-box-color" [ngClass]="[form.controls[formControllerName].invalid && form.controls[formControllerName].touched ? 'error-box' : 'normal-box']"  pInputText [formControlName]="formControllerName" [placeholder]="placeholder" [required]="isFieldMandatory">
+        <span class="p-input-icon-right">
+            <i *ngIf="type === 'password'" class="pi pi-eye tw-cursor-pointer" (click)="changeFormControllerType(true)"></i>
+            <i *ngIf="type !== 'password' && formControllerName === 'userPassword'" class="pi pi-eye-slash tw-cursor-pointer" (click)="changeFormControllerType(false)"></i>
+            <input [type]="type" class="tw-border-box-color" [ngClass]="[form.controls[formControllerName].invalid && form.controls[formControllerName].touched ? 'error-box' : 'normal-box']"  pInputText [formControlName]="formControllerName" [placeholder]="placeholder" [required]="isFieldMandatory">
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-text-field/configuration-text-field.component.html:13]: quite didn't understand what are we trying to do here

File: src/app/shared/components/helper/app-landing-page-body/app-landing-page-body.component.html [hunk 9]
```diff
@@ -13,7 +13,7 @@
         <iframe class= "tw-rounded-8-px" width="720" height="420" [src]="embedVideoUrl" title="{{brandingConfig.brandName}} Sage Intacct Integration" frameborder="0"></iframe>
     </div>    
 
-    <div class="tw-flex tw-justify-center tw-mt-8 tw-mb-14">
+    <div class="tw-flex tw-justify-center tw-mt-20 tw-mb-14">
```
Comment (ashwin1111) [inline src/app/shared/components/helper/app-landing-page-body/app-landing-page-body.component.html:16]: mt-80-px, use px

File: src/app/shared/components/helper/onboarding-done/onboarding-done.component.html [hunk 10]
```diff
@@ -2,19 +2,20 @@
 <div class="tw-absolute tw-bg-header-2 tw-h-110-px tw-w-[100%] tw-z-[-1]"></div>
 <div class="tw-flex tw-justify-center tw-pr-74-px tw-pl-74-px tw-pt-24 tw-pb-20">
   <div class="onboarding-done--contents">
-    <div class="tw-pt-34-px">
-      <img src="assets/gifs/success-check.gif" class="onboarding-done--align-center" alt="Onboarding done" width="120px"
-        height="120px">
+    <div class="tw-pt-34-px tw-mb-30-px">
+      <!-- <svg-icon-sprite class="onboarding-done--align-center" src="onboarding-done" width="120px"
```
Comment (ashwin1111) [inline src/app/shared/components/helper/onboarding-done/onboarding-done.component.html:6]: remove commented code

File: src/app/integrations/sage300/sage300-shared/sage300-advanced-settings/sage300-advanced-settings.component.scss (hunk 3 above)
Comment (DhaaraniCIT) [reply src/app/integrations/sage300/sage300-shared/sage300-advanced-settings/sage300-advanced-settings.component.scss:2]: yes, it should be for top and bottom as well

Comment (ashwin1111) [reply src/app/integrations/sage300/sage300-shared/sage300-advanced-settings/sage300-advanced-settings.component.scss:2]: pt should be 0

File: src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.ts [hunk 11]
```diff
@@ -51,6 +51,16 @@ export class ConfigurationImportFieldComponent implements OnInit {
...
 
+  getDynamicWidth(index: number): string {
+    const length = 260 - (index * 20);
+    return `tw-w-${length}-px`;
+  }
+
+  getDynamicPadding(index: number): string{
+    const padding = index * 20;
+    return `tw-pl-${padding}-px`;
+  }
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.ts:62]: remove unused func

**Review by ashwin1111 — APPROVED** [review]

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**